src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -110,11 +138,64 @@ private void processTripQueryParams(MonitoredTrip monitoredTrip, Request req) {
 
     @Override
     MonitoredTrip preUpdateHook(MonitoredTrip monitoredTrip, MonitoredTrip preExisting, Request req) {
-        preCreateOrUpdateChecks(monitoredTrip, req);
+        // Wait for any existing CheckMonitoredTrip jobs to complete before proceeding
+        if (monitoredTripLocks.containsKey(monitoredTrip)) {
+            int maxWaitTimeMillis = 4000;
+            int timeWaitedMillis = 0;
+            do {
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    logMessageAndHalt(
+                        req,
+                        HttpStatus.INTERNAL_SERVER_ERROR_500,
+                        "The trip analyzer prevented the update of this trip. Please try again."
+                    );
+                }
+                timeWaitedMillis += 500;
 
-        // TODO: Update itinerary existence record when updating a trip.
+                // if the lock has been released, exit this wait loop
+                if (!monitoredTripLocks.containsKey(monitoredTrip)) break;
+            } while (timeWaitedMillis <= maxWaitTimeMillis);
+        }
 
-        return monitoredTrip;
+        // If a lock still exists, prevent the update
+        if (monitoredTripLocks.containsKey(monitoredTrip)) {
+            logMessageAndHalt(
+                req,
+                HttpStatus.INTERNAL_SERVER_ERROR_500,
+                "The trip analyzer prevented the update of this trip. Please try again."
```

Comment:
This message could be placed into a constant.

Reply:
Refactored in 6948fab3bf5ead8b87a4b47676a38fe9a6e1691f